src/libs/ui/mainwindow.h

```diff
@@ -78,10 +79,13 @@ public slots:
 private slots:
     void applySettings();
     void openDocset(const QModelIndex &index);
+    void openAndFocusDocset(const QModelIndex &index);
```

Comment:
Let's instead of adding a new method, change `openDocset` to

```cpp
void openDocset(const QModelIndex &index, bool focusWebView = true);
```

Reply:
Hmm, looks like this isn't going to work actually - Qt allows a slot to ignore parameters provided by a signal, but it doesn't let you do the opposite:

```cpp
// mainwindow.h
void openDocset(const QModelIndex &index, bool focusWebView = true);
```

```cpp
// mainwindow.cpp
connect(ui->treeView, &QTreeView::clicked, this, &MainWindow::openDocset);
```

```
error: static assertion failed: The slot requires more arguments than the signal provides.
         Q_STATIC_ASSERT_X(int(SignalType::ArgumentCount) >= int(SlotType::ArgumentCount),
```

Doesn't work with an override of `void openDocset(const QModelIndex &index)` either, or explicitly setting signatures using `SLOT()`/`SIGNAL()` :thinking:

Reply:
Indeed, then there's no escape from two slots :)